6 add schemaorg data to head of view page

nbodony opened the pull request:
Json-ld string is placed in head of view.html and can be found at /ld/dataset. It only includes the information that I was aware of and thought to include so it's definitely not complete but I think it works how it's supposed to and is placed everywhere it needs to be.

/ld/piglife: 
![image](https://github.com/user-attachments/assets/dd8fc452-18b0-4c46-beaa-09e25d05dbc8)

/ld/slamdataset:   (the list of authors is very long)
![image](https://github.com/user-attachments/assets/2811a0f8-5744-4017-b649-bf03e8cce1c3)

/view/slamdataset:
![image](https://github.com/user-attachments/assets/032ca482-4e1a-4769-b53f-2c99d8f5fce6)

/view/piglife
![image](https://github.com/user-attachments/assets/9383b211-306e-40bc-9f27-cf41764e247f)

nbodony committed:
Initial schema string creation

creates an initial and very incomplete schema string which can reached at /ld/dataset

nbodony committed:
remove extra line in identifier

nbodony committed:
remove extra line at start of string

nbodony committed:
many schema string changes

make schema a python dict
convert dict to json string
make variable list of creator blocks
move script to view.html
pass schema_string to view.html head

nbodony committed:
removed testing thing

nbodony committed:
conflict

robkooper commented on app.py:
```diff
@@ -37,6 +38,57 @@ def load_data():
+
+
+def create_json_ld(dataset):
+    data = load_data()
```
lets be safe and make sure dataset is in data

robkooper commented on app.py:
```diff
@@ -37,6 +38,57 @@ def load_data():
+def create_json_ld(dataset):
+    data = load_data()
+    schema_dict = {
+      "@context":"https://schema.org/",
```
be consistent to have a space (preferred) or not after the colon.

robkooper commented on app.py:
```diff
@@ -37,6 +38,57 @@ def load_data():
+      "sameAs": "",
+      "version": "",
+      "isAccessibleForFree": True,
+      "keywords": "{{ keywords }}",
```
this should be an array.

robkooper commented on app.py:
```diff
@@ -37,6 +38,57 @@ def load_data():
+      },
+      "citation": "{{ citation }}",
+      "creator": [],
+      "provider": {
```
if not filled out, just comment them out for now, same below.

robkooper commented on app.py:
```diff
@@ -37,6 +38,57 @@ def load_data():
+
+def create_json_ld(dataset):
+    data = load_data()
+    schema_dict = {
```
lets just json_ld instead of schema. Schema is very generic, json_ld describes what the data is.

nbodony committed:
app.py changes + keywords

Removed unnecessary arrays and added one necessary array

added keywords with correct spacing after colon

commented out unused lines in json_ld_dict

replaced 'schema' with 'json_ld'

added missing spaces after colons

nbodony committed:
return empty dict if dataset not in data

robkooper commented on app.py:
```diff
@@ -37,6 +38,60 @@ def load_data():
+      #"sameAs": "",
+      #"version": "",
+      "isAccessibleForFree": True,
+      "keywords": [],
```
shouldn't this be
```suggestion
      "keywords": data[dataset]['keywords'],
```
one safe way to do this is:
```
      "keywords": data[dataset].get('keywords', []),
```

nbodony committed:
Merge branch 'main' into 6-add-schemaorg-data-to-head-of-view-page

nbodony committed:
better keywords

nbodony committed:
app.py fix
